File: Software/app/src/main/java/hr/foi/rampu/fridgium/database/MjernaJedinicaDAO.kt
package hr.foi.rampu.fridgium.database

import hr.foi.rampu.fridgium.entities.MjernaJedinica
import hr.foi.rampu.fridgium.entities.Namirnica
import hr.foi.rampu.fridgium.entities.NamirnicaRecepta
import java.sql.ResultSet
const val mj : String ="mjerna_jedinica"

class MjernaJedinicaDAO {

    fun getMjerneJedinice(): MutableList<MjernaJedinica> {
        val sql: String = "SELECT * FROM heroku_d1561a1a0615483.$mj"
        val vrijednosti : MutableList<MjernaJedinica> = pretvoriVrijednosti(Connector.executeSearchQuery(sql))
        return vrijednosti
    }

    fun getMjernaJedinica(id: Int): MjernaJedinica {
        val sql: String = "SELECT * FROM heroku_d1561a1a0615483.$mj WHERE id=$id"
        val vrijednost : MjernaJedinica = pretvoriVrijednost(Connector.executeSearchQuery(sql))
        return vrijednost
    }

    fun pretvoriVrijednosti(resultSet: ResultSet): MutableList<MjernaJedinica>{
        val popis : MutableList<MjernaJedinica> = arrayListOf()
        while(resultSet.next()){
            val id: Int  = resultSet.getInt("id")
            val naziv: String = resultSet.getString("naziv")
            val MjernaJedinica = MjernaJedinica(id,naziv)
            popis.add(MjernaJedinica)
        }
        return popis

    }

    fun pretvoriVrijednost(resultSet: ResultSet): MjernaJedinica{
            val id: Int  = resultSet.getInt("id")
            val naziv: String = resultSet.getString("naziv")
            val MjernaJedinica = MjernaJedinica(id,naziv)
            return MjernaJedinica

    }
}

File: Documentation/REST Aplikacija/DAO/mjernajedinicaDAO.js
const Baza = require("../baza.js");

class mjernajedinicaDAO{

    constructor(){
        this.baza = new Baza();
    }

    dajSve = async function () {
        this.baza.spojiSeNaBazu();
        let sql = "SELECT * FROM heroku_d1561a1a0615483.mjerna_jedinica"
        var podaci = await this.baza.izvrsiUpit(sql, [])
        this.baza.zatvoriVezu();
        return podaci;
    }

    daj = async function (id) {
        this.baza.spojiSeNaBazu();
        try{
            let sql = "SELECT * FROM heroku_d1561a1a0615483.mjerna_jedinica WHERE id=?"
            var podaci = await this.baza.izvrsiUpit(sql, [id])
        }finally{
            this.baza.zatvoriVezu();
        }
        if(podaci.length == 1)
            
			return podaci[0];
		else 
			return null;
    }

}

module.exports = mjernajedinicaDAO;